build.sbt

```diff
@@ -423,29 +419,35 @@ lazy val crossVersionSharedSources: Seq[Setting[_]] =
   Seq(Compile, Test).map { sc =>
     (sc / unmanagedSourceDirectories) ++= {
       (sc / unmanagedSourceDirectories).value.map { dir: File =>
-        CrossVersion.partialVersion(scalaBinaryVersion.value) match {
+        CrossVersion.partialVersion(scalaVersion.value) match {
           case Some((major, minor)) =>
             new File(s"${dir.getPath}_$major.$minor")
           case None =>
             sys.error("couldn't parse scalaBinaryVersion ${scalaBinaryVersion.value}")
         }
       }
     }
-  }
+  } ++ Seq(
+    Compile / unmanagedSourceDirectories ++= scalaVersionSpecificFolders("main",
+                                                                         baseDirectory.value,
+                                                                         scalaVersion.value
+    ),
+    Test / unmanagedSourceDirectories ++= scalaVersionSpecificFolders("test", baseDirectory.value, scalaVersion.value)
+  )
 
 lazy val scalaMacroDependencies: Seq[Setting[_]] = Seq(
-  libraryDependencies += scalaOrganization.value % "scala-reflect" % scalaVersion.value % "provided"
+  libraryDependencies ++= {
+    if (scalaVersion.value.startsWith("3")) Seq.empty
+    else Seq(scalaOrganization.value % "scala-reflect" % scalaVersion.value % "provided")
+  }
 )
 
 lazy val commonScalacOptions = Def.setting(
   (CrossVersion.partialVersion(scalaVersion.value) match {
     case Some((2, v)) if v >= 13 =>
       Seq()
     case _ =>
-      Seq(
-        "-Yno-adapted-args",
-        "-Xfuture"
-      )
+      Seq()
   }) ++ Seq(
```

Comment:
This part can be removed now it seems, both match cases are `Seq()`.

Reply:
Yeah, again it is somehow a leftover during the conversion process. I'd like to go towards using `sbt-tpolecat` with some exceptions (like using vars)) in the near future